src/python/pants/rules/core/list_targets.py

```diff
@@ -1,35 +1,40 @@
-# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
-from typing import Union
+from typing import Dict, cast
 
+from pants.build_graph.address import Address
 from pants.engine.addressable import Addresses
 from pants.engine.console import Console
 from pants.engine.goal import Goal, GoalSubsystem, LineOriented
-from pants.engine.legacy.graph import HydratedTargets
 from pants.engine.rules import goal_rule
 from pants.engine.selectors import Get
+from pants.engine.target import DescriptionField, ProvidesField, Targets
 
 
 class ListOptions(LineOriented, GoalSubsystem):
-    """Lists all targets matching the target specs."""
+    """Lists all targets."""
 
-    name = "list"
+    name = "list-v2"
```

Comment:
I wanted to call it `list2`, but that will cause failures because we rename `goal2` to `goal` when V2 is exclusively used, so we would now have two goals named `list`.